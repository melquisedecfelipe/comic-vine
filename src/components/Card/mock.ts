import { CardProps } from '.'

export default {
  character: {
    aliases: 'The Flame Alchemist Colonel Mustang General Mustang',
    detail: 'https://comicvine.gamespot.com/api/character/4005-49318/',
    birth: 'Sep 29, 1886',
    id: 49318,
    images: {
      icon: 'https://comicvine.gamespot.com/a/uploads/square_avatar/11117/111178336/5581737-roy_mustang_by_norygunz-d4rd8xl.png',
      screen:
        'https://comicvine.gamespot.com/a/uploads/screen_medium/11117/111178336/5581737-roy_mustang_by_norygunz-d4rd8xl.png',
      small:
        'https://comicvine.gamespot.com/a/uploads/scale_small/11117/111178336/5581737-roy_mustang_by_norygunz-d4rd8xl.png'
    },
    name: 'Roy Mustang',
    realName: 'Roy Mustang',
    deck: 'Roy Mustang, the Flame Alchemist, is the Colonel in Central and has jurisdiction over most alchemist there. Known for his heroism in Ishval and his amazing skill with his Ignition Cloth Gloves, Roy has great ambitions to become the next Fuhrer of Amestris.',
    description:
      '<h4>Background/Basic Information</h4><p>The Flame Alchemist and Hero of Ishval, Roy Mustang is the colonel of the Amestrian Military and a State Alchemist. Roy was one of the few to make it through the Ishvalan Civil War unscathed, though the atrocities committed in the great struggle are part of what drove his ambition to become the Fuhrer of Amestris.</p><p>His closest friends are <a href="/maes-hughes/4005-50742/" data-ref-id="4005-50742">Maes Hughes </a>and <a href="/riza-hawkeye/4005-52046/" data-ref-id="4005-52046">Lieutenant Riza Hawkeye</a>, the former of which was murdered by <a href="/envy/4005-50301/" data-ref-id="4005-50301">Envy</a> after he discovered the Homunculi\'s plans. As a result of Hughes\' death, the Colonel went on a vendetta to find the killer, culminating in a violent confrontation with Envy as Roy wreaked vengeance upon the murderous homunculus.</p><p>After <a href="/pride/4005-50303/" data-ref-id="4005-50303">Pride</a> and <a href="/wrath/4005-50262/" data-ref-id="4005-50262">Wrath</a> forced him to open the portal and see the Truth, Roy was used as one of <a href="/father/4005-50299/" data-ref-id="4005-50299">Father</a>\'s sacrifices in his grand scheme to absorb all the souls in Amestris and swallow God. Roy was rendered blind as a result. After Father\'s plan was unraveled through the efforts of <a href="/hohenheim/4005-51726/" data-ref-id="4005-51726">Van Hohenheim</a>, Roy was one of those who took place in the final battle. Afterwards, he went on to become the Fuhrer and his vision was restored by a remaining Philosopher\'s Stone.</p><h4>Powers</h4><p>As the Flame Alchemist, Roy makes use of Ignition Cloth Gloves to create a spark when he snaps; by then manipulating the oxygen density in the air using alchemy, he can then generate flames. His flame alchemy ranges from great bursts and explosions to very precise and surgical fire.</p><p>If his special gloves are wet, however, he cannot create the sparks necessary to produce flames, hence being "useless on rainy days". Aside from these, he also knows the standard form of alchemy.</p>',
    site: 'https://comicvine.gamespot.com/roy-mustang/4005-49318/',
    slug: '49318-roy-mustang'
  }
} as CardProps
