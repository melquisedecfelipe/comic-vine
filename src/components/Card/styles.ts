import media from 'styled-media-query'

import styled, { css, DefaultTheme } from 'styled-components'

type WrapperProps = {
  $isVertical: boolean
}

const wrapperModifiers = {
  isVertical: (theme: DefaultTheme) => css`
    grid-template-columns: 1fr;
    border: 0;
    padding: 0;
    padding-top: ${theme.spacings.medium};

    > section {
      position: absolute;
      bottom: 0;
      left: 0;
      background: ${theme.colors.black};
      margin: 0;

      & h3 {
        color: ${theme.colors.white};
        margin: ${theme.spacings.xsmall};
        -webkit-line-clamp: 1 !important;
      }
    }
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, $isVertical }) => css`
    display: flex;
    flex-direction: column;
    position: relative;
    height: 100%;
    background: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.greyLight};
    padding: ${theme.spacings.medium} 0;
    transition: ${theme.transition.default};

    & img {
      transition: ${theme.transition.default};
    }

    &:hover {
      & img {
        filter: grayscale(1);
      }
    }

    ${media.greaterThan('medium')`
      display: grid;
      grid-template-columns: 175px 1fr;
    `}

    ${$isVertical && wrapperModifiers.isVertical(theme)}
  `}
`

export const ImageWrapper = styled.aside`
  position: relative;
  height: 100%;
  width: 175px;

  > button {
    position: absolute;
    top: 0;
    right: 0;

    > svg {
      width: 20px;
      height: 20px;
    }
  }
`

export const Section = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    cursor: pointer;

    ${media.greaterThan('medium')`
      margin-left: ${theme.spacings.medium};
    `}

    & small {
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.grey};
      margin: ${theme.spacings.xxsmall} 0;
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
    }

    & h3 {
      font-family: ${theme.font.familyBarlow};
      font-weight: ${theme.font.semiBold};
      font-size: ${theme.font.sizes.large};
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
    }

    & p {
      font-size: ${theme.font.sizes.medium};
      color: ${theme.colors.grey};
      margin: ${theme.spacings.small} 0;
      overflow: hidden;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 5;
      -webkit-box-orient: vertical;
    }
  `}
`
