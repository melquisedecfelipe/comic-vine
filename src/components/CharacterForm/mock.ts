import { CharacterEdit } from 'types/characters'

export default {
  aliases: 'Jessica Drew Arachne Ariadne Hyde Hunter',
  birth: '1924-09-07',
  gender: 'Female',
  id: 1453,
  name: 'Spider-Woman',
  realName: 'Jessica Miriam Drew'
} as CharacterEdit
