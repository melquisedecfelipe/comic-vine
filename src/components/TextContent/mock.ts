import { TextContentProps } from '.'

export default {
  content:
    '<h2>Origin</h2><figure data-align="right" data-size="small" data-ref-id="1300-2806772" data-img-src="https://static.comicvine.com/uploads/original/11/117480/2806772-punisher_porfolio_003.jpg" data-ratio="1.4773060029283" data-width="683" data-embed-type="image" style="width: 683px"><a class="fluid-height" style="padding-bottom:147.7%" href="https://static.comicvine.com/uploads/original/11/117480/2806772-punisher_porfolio_003.jpg" data-ref-id="1300-2806772"><img alt="Beginnings" src="https://comicvine.gamespot.com/a/uploads/scale_small/11/117480/2806772-punisher_porfolio_003.jpg" srcset="https://comicvine.gamespot.com/a/uploads/original/11/117480/2806772-punisher_porfolio_003.jpg 683w, https://comicvine.gamespot.com/a/uploads/scale_medium/11/117480/2806772-punisher_porfolio_003.jpg 649w, https://comicvine.gamespot.com/a/uploads/scale_small/11/117480/2806772-punisher_porfolio_003.jpg 320w" sizes="(max-width: 320px) 100vw, 320px" data-width="320"></a><figcaption>Beginnings</figcaption></figure><p>A decorated US <a href="https://comicvine.gamespot.com/united-states-marine-corps/4060-44416/" data-ref-id="4060-44416">Marine</a> excelling as a soldier in every way, Frank Castle was a force to be reckoned with during the <a href="https://comicvine.gamespot.com/vietnam-war/4015-56024/" data-ref-id="4015-56024">Vietnam War</a> but also a dedicated family man. After a particularly close call in the final years of the war, he resigned to a <a href="https://comicvine.gamespot.com/united-states-special-forces/4060-55045/" data-ref-id="4060-55045">Special Forces</a> instructor post in upstate <a href="https://comicvine.gamespot.com/new-york-state/4020-56310/" data-ref-id="4020-56310">New York</a> so he could be closer to his wife <a href="/maria-castle/4005-63156/" data-ref-id="4005-63156">Maria </a>and two children, <a href="/lisa-castle/4005-63158/" data-ref-id="4005-63158">Lisa </a>and <a href="/francis-castle-jr/4005-63159/" data-ref-id="4005-63159">Francis</a>. Disaster struck one day, years later, when Frank and his family went for a picnic in Central Park only to witness a mob execution on a high profile member of the mafia. The mob fired on his family, killing them all before his eyes and thus destroying everything he had lived for.</p><p>Unable to move on and failed by a corrupt criminal justice system, Frank Castle sought revenge, prompting him to "disappear" just as he was to receive the Presidential Medal of Freedom. As he resurfaced, he was the vigilante now known as the Punisher, wearing Kevlar body armor with a large skull imprinted on his chest and armed with a variety of weapons. Frank not only vowed to have revenge for his family, but also dedicated the rest of his life to fighting crime.</p><h2>Creation</h2><figure data-align="right" data-size="small" data-ref-id="1300-2806792" data-img-src="https://static.comicvine.com/uploads/original/11/117480/2806792-weinspidey12.jpg" data-ratio="1.4833333333333" data-width="600" data-embed-type="image" style="width: 600px"><a class="fluid-height" style="padding-bottom:148.3%" href="https://static.comicvine.com/uploads/original/11/117480/2806792-weinspidey12.jpg" data-ref-id="1300-2806792"><img alt="Punisher saving the day." src="https://comicvine.gamespot.com/a/uploads/scale_small/11/117480/2806792-weinspidey12.jpg" srcset="https://comicvine.gamespot.com/a/uploads/original/11/117480/2806792-weinspidey12.jpg 600w, https://comicvine.gamespot.com/a/uploads/scale_medium/11/117480/2806792-weinspidey12.jpg 480w, https://comicvine.gamespot.com/a/uploads/scale_small/11/117480/2806792-weinspidey12.jpg 431w" sizes="(max-width: 431px) 100vw, 431px" data-width="431"></a><figcaption>Punisher saving the day.</figcaption></figure><p>Frank Castle debuted in 1974\'s <a href="/the-amazing-spider-man-the-punisher-strikes-twice/4000-105443/" data-ref-id="4000-105443"> The Amazing Spider-Man #129</a> as the vigilante assassin hired by <a href="/jackal/4005-10974/" data-ref-id="4005-10974"> Jackal</a> to kill <a href="/spider-man/4005-1443/" data-ref-id="4005-1443"> Spider-Man</a> for the apparent murder of <a href="/norman-osborn/4005-58812/" data-ref-id="4005-58812"> Norman Osborn</a>. As differences between Punisher and Jackal came to a head, Punisher realized that he was being tricked into killing Spider-Man and that our friendly neighborhood web-slinger was not in fact a criminal or murderer. From then on, audiences came to understand Punisher\'s origin and that he was not a villain and he then returned to aid Spider-Man when old war buddy turned insane war criminal Hitman was hired to kill the web head. Created by writer <a href="../../gerry-conway/26-40871/" rel="nofollow "> Gerry Conway</a>, artists <a href="/marvel-visionaries-john-romita-sr/4050-41312/" data-ref-id="4050-41312"> John Romita</a> <a href="/marvel-visionaries-john-romita-sr/4050-41312/" data-ref-id="4050-41312"> Sr</a> and <a href="../../ross-andru/26-29364/" rel="nofollow "> Ross Andru</a> and Stan Lee having suggested the Punisher name, Conway originally designed Frank Castle to operate as a villain, but as the story developed he became an antihero. Conway has said his biggest influences in creating the character were Don Pendleton\'s Executioner: <a href="../../mack-bolan/29-54652/" rel="nofollow "> Mack Bolan</a>.</p>'
} as TextContentProps
