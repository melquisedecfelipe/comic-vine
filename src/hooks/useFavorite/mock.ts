export default [
  {
    aliases:
      'James Howlett Logan Agent Ten Black Dragon Brother Xavier Captain Canada Captain Terror Death Emilio Garra Experiment X Fist of Legend Hooded Man Jim Logan John Logan Mutate #9601 Patch Peter Richards Revolto the Clown Weapon X',
    detail: 'https://comicvine.gamespot.com/api/character/4005-1440/',
    birth: null,
    id: 1440,
    images: {
      icon: 'https://comicvine.gamespot.com/a/uploads/square_avatar/5/57023/7469590-wolverinerb.jpg',
      screen:
        'https://comicvine.gamespot.com/a/uploads/screen_medium/5/57023/7469590-wolverinerb.jpg',
      small:
        'https://comicvine.gamespot.com/a/uploads/scale_small/5/57023/7469590-wolverinerb.jpg'
    },
    name: 'Wolverine',
    realName: 'James Howlett',
    deck: 'A long-lived mutant with the rage of a beast and the soul of a Samurai, James "Logan" Howlett\'s once mysterious past is filled with blood, war, and betrayal. Possessing an accelerated healing factor, keenly enhanced senses, and bone claws in each hand (along with his skeleton) that are coated in adamantium; Wolverine is, without question, the ultimate weapon.',
    description: `<p>The man who would be known as the superhero named Wolverine was born as James Howlett in the mid 1880s in Alberta, <a href="../../canada/34-55868/" rel="nofollow ">Canada</a>. He was the second and sickly child to <a href="../../elizabeth-howlett/29-44325/" rel="nofollow ">Elizabeth</a> and <a href="../../howlett-senior/29-44326/" rel="nofollow ">John Howlett Sr.</a> His mother was ill and, in the early 1900s, a young Irish girl named <a href="../../rose/29-44342/" rel="nofollow ">Rose O'Hara</a> was brought to the Howlett estate to be a friend and a caretaker to young James. Together they befriended a young boy named <a href="../../dog-logan/29-44327/" rel="nofollow ">Dog</a>, son of the grounds keeper, <a href="../../thomas-logan/29-44323/" rel="nofollow ">Thomas Logan</a>. Dog tried to form a normal bond between the two, but due to his father’s resentment and drunken abuse, he grew to resent his station in life; so much so that one day he makes undue advances on Rose, which are reported by young James to his father. This whole chain of events leads to <a href="../../thomas-logan/29-44323/" rel="nofollow ">Thomas Logan</a> being sacked from his position as groundskeeper, following which he goes seeking for <a href="/elizabeth-howlett/4005-44325/" data-ref-id="4005-44325">Elizabeth Howlett</a>, trying to convince her to leave with him. This leads to an altercation that results in the death of <a href="/john-howlett/4005-4657/" data-ref-id="4005-4657">John Howlett</a>. The shock of seeing his father murdered in front of him leads to the manifestation of young James’ powers and he stabs <a href="/thomas-logan/4005-44323/" data-ref-id="4005-44323">Thomas </a>with his newly drawn claws, also slashing Dog across the face before fainting. He and Rose flee the house and make their way through the harsh Canadian wilderness with James near-catatonic and Rose having to facilitate their transportation, while <a href="/dog-logan/4005-44327/" data-ref-id="4005-44327">Dog</a> tells the police that Rose was the killer.</p>`,
    site: 'https://comicvine.gamespot.com/wolverine/4005-1440/',
    slug: '1440-wolverine'
  },
  {
    aliases:
      'Kal-El Clark Kent Clark Jerome Kent Clark Joseph Kent Gangbuster Nightwing Jordan Elliot Superboy Superman-Prime The Action Ace The Big Blue Boy Scout The Last Son of Krypton The Man of Might The Man of Steel The Man of Tomorrow Marc Costa Nembo Kid Aksel  Archie Clayton Clark White God of Strength Flying Man Skyman',
    detail: 'https://comicvine.gamespot.com/api/character/4005-1807/',
    birth: 'Jun 18, 1971',
    id: 1807,
    images: {
      icon: 'https://comicvine.gamespot.com/a/uploads/square_avatar/13/132327/6507037-28872490_1638064799604695_1250122498385004714_n.jpg',
      screen:
        'https://comicvine.gamespot.com/a/uploads/screen_medium/13/132327/6507037-28872490_1638064799604695_1250122498385004714_n.jpg',
      small:
        'https://comicvine.gamespot.com/a/uploads/scale_small/13/132327/6507037-28872490_1638064799604695_1250122498385004714_n.jpg'
    },
    name: 'Superman',
    realName: 'Kal-El / Clark Joseph Kent',
    deck: "Rocketed to Earth as an infant from the doomed planet Krypton, Kal-El was adopted by the loving Kent family and raised in America's heartland as Clark Kent. Using his immense solar-fueled powers, he became Superman to defend mankind against all manner of threats while championing truth, justice and the American way!",
    description: `<p>Although Superman's origin has been retold many times, the story has remained somewhat consistent throughout. Superman's story tells of his birth as Kal-El to renowned scientist <a href="/jor-el/4005-9561/" data-ref-id="4005-9561">Jor-El</a> and his wife <a href="/lara-lor-van/4005-9562/" data-ref-id="4005-9562">Lara</a> on the alien planet <a href="/krypton/4020-30670/" data-ref-id="4020-30670">Krypton</a>. When Jor-El discovers that the planet is going to explode, he puts Kal-El in a spaceship heading to the planet Earth, which crashes in a field in the state of <a href="/kansas/4020-55854/" data-ref-id="4020-55854">Kansas</a>.</p>`,
    site: 'https://comicvine.gamespot.com/superman/4005-1807/',
    slug: '1807-superman'
  }
]
