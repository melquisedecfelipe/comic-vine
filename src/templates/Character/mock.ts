import { CharacterProps } from '.'

export default {
  aliases:
    'Peter Parker\nCaptain Universe\nDusk\nHornet\nProdigy\nRicochet\nBombastic Bag-Man\nPoison\nPestilence\nSpidey\nWallcrawler\nWebslinger\nWeb-Head\nYour Friendly Neighborhood Spider-Man\nSpider-Kid',
  detail: 'https://comicvine.gamespot.com/api/character/4005-1443/',
  birth: 'Oct 14, 1962',
  id: 1443,
  images: {
    icon: 'https://comicvine.gamespot.com/a/uploads/square_avatar/12/124259/8126579-amazing_spider-man_vol_5_54_stormbreakers_variant_textless.jpg',
    screen:
      'https://comicvine.gamespot.com/a/uploads/screen_medium/12/124259/8126579-amazing_spider-man_vol_5_54_stormbreakers_variant_textless.jpg',
    small:
      'https://comicvine.gamespot.com/a/uploads/scale_small/12/124259/8126579-amazing_spider-man_vol_5_54_stormbreakers_variant_textless.jpg'
  },
  name: 'Spider-Man',
  realName: 'Peter Benjamin Parker',
  deck: 'Peter Parker was bitten by a radioactive spider as a teenager, granting him spider-like powers. After the death of his Uncle Ben, Peter learned that "with great power, comes great responsibility." Swearing to always protect the innocent from harm, Peter Parker became Spider-Man.',
  description: `<p>Peter Benjamin Parker was born to C.I.A. agents <a href="../../richard-parker/29-15261/" rel="nofollow ">Richard</a> and <a href="../../mary-parker/29-15262/" rel="nofollow ">Mary Parker</a>, who were killed when Peter was very young. After the death of his parents, Peter was raised by his <a href="../../uncle-ben/29-3114/" rel="nofollow ">Uncle Ben</a> and <a href="../../aunt-may/29-1780/" rel="nofollow ">Aunt May</a> in a <a href="/forest-hills-residence/4020-57247/" data-ref-id="4020-57247">modest house</a> in Forest Hills, <a href="/new-york-city/4020-41183/" data-ref-id="4020-41183">New York</a>. While still in <a href="/midtown-high-school/4020-55690/" data-ref-id="4020-55690">High School</a>, Peter attended a science exhibition about radiology where he was bitten by a <a href="/radioactive-spider/4055-56442/" data-ref-id="4055-56442">radioactive spider</a>, granting him the proportionate strength and agility of a spider as well as a " <a href="/spider-sense/4015-56106/" data-ref-id="4015-56106">Spider-Sense</a>" that warns him of nearby danger. In order to test his new abilities – and earn some cash, Peter participated in a wrestling challenge against <a href="/crusher-hogan/4005-15480/" data-ref-id="4005-15480">Crusher Hogan</a>. He easily won the challenge and also gained the attention of the media. Afterward while backstage, Peter saw a burglar run past him but did nothing to stop him as it wasn’t his problem. Later that night when Peter arrived home, he was told by a policeman that his Uncle Ben had been murdered by someone who broke into their house. The cop mentioned they had tracked the killer to a warehouse. In his anger, Peter put on his wrestling costume and went after the murderer himself. After arriving at the warehouse and easily defeating Uncle Ben’s killer, Peter saw that it was the same <a href="/the-burglar/4005-15279/" data-ref-id="4005-15279">burglar</a> he didn’t stop at the arena… thus learning that "<a href="/with-great-power-comes-great-responsibility/4015-40616/" data-ref-id="4015-40616">with great power comes great responsibility</a>".</p>`,
  site: 'https://comicvine.gamespot.com/spider-man/4005-1443/',
  slug: '1443-spider-man',
  origin: 'Human',
  publisher: 'Marvel'
} as CharacterProps
