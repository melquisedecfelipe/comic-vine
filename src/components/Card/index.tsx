import { useRouter } from 'next/dist/client/router'
import Image from 'next/image'

import { Heart as HeartIcon } from '@styled-icons/boxicons-regular/Heart'
import { Heart as HeartFillIcon } from '@styled-icons/boxicons-solid/Heart'

import Button from 'components/Button'

import * as S from './styles'

import { Character } from 'types/characters'

export type CardProps = {
  character: Character
  handleAddFavorite: (character: Character) => void
  handleRemoveFavorite: (id: number) => void
  isFavorite: (id: number) => boolean
  isVertical?: boolean
}

const Card = ({
  character,
  handleAddFavorite,
  handleRemoveFavorite,
  isFavorite,
  isVertical = false
}: CardProps) => {
  const { push } = useRouter()

  return (
    <S.Wrapper $isVertical={isVertical}>
      <S.ImageWrapper>
        <Image
          alt={character.name}
          src={character.images.small}
          width={175}
          height={250}
          layout="responsive"
          objectFit="cover"
        />

        {isFavorite(character.id) ? (
          <Button
            aria-label="Remove Favorite"
            onClick={() => handleRemoveFavorite(character.id)}
          >
            <HeartFillIcon />
          </Button>
        ) : (
          <Button
            aria-label="Add Favorite"
            onClick={() => handleAddFavorite(character)}
          >
            <HeartIcon />
          </Button>
        )}
      </S.ImageWrapper>

      <S.Section
        aria-label="Character"
        role="link"
        onClick={() => push(`character/${character.slug}`)}
      >
        <span>
          {!isVertical && character.birth && <small>{character.birth}</small>}
          <h3>{character.name}</h3>
        </span>

        {!isVertical && character.deck && <p>{character.deck}</p>}
        {!isVertical && <small>Aliases: {character.aliases}</small>}
      </S.Section>
    </S.Wrapper>
  )
}

export default Card
