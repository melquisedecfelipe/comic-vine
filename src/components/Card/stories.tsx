import { Story, Meta } from '@storybook/react/types-6-0'

import Card, { CardProps } from '.'
import mock from './mock'

export default {
  title: 'Card',
  component: Card,
  args: mock,
  parameters: {
    nextRouter: {
      path: '/character/[slug]',
      asPath: `/character/${mock.href}`,
      query: {
        slug: mock.href
      }
    }
  }
} as Meta

export const Default: Story<CardProps> = args => (
  <div style={{ maxWidth: 500 }}>
    <Card {...args} />
  </div>
)
