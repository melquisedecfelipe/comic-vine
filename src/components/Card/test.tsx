import userEvent from '@testing-library/user-event'

import { render, screen } from 'utils/testUtils'

import Card, { CardProps } from '.'
import mock from './mock'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const isFavorite = () => false
const handleAddFavorite = jest.fn()
const handleRemoveFavorite = jest.fn()
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push
}))

const props = {
  character: mock.character,
  isFavorite,
  handleAddFavorite,
  handleRemoveFavorite
} as CardProps

describe('<Card />', () => {
  it('should render the card', () => {
    const { container } = render(<Card {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('should render the card vertical', () => {
    render(<Card {...props} isVertical />)

    expect(
      screen.getByRole('link', { name: /character/i }).parentElement
    ).toHaveStyle({
      'grid-template-columns': '1fr',
      border: 0
    })
  })

  it('should redirect on card click', () => {
    render(<Card {...props} />)

    userEvent.click(screen.getByRole('link', { name: /character/i }))

    expect(push).toHaveBeenCalledTimes(1)
  })

  it('should favorite character on button click', () => {
    render(<Card {...props} />)

    userEvent.click(screen.getByRole('button', { name: /add favorite/i }))

    expect(handleAddFavorite).toHaveBeenCalledTimes(1)
  })

  it('should remove favorite character on button click', () => {
    render(<Card {...props} isFavorite={() => true} />)

    userEvent.click(screen.getByRole('button', { name: /remove favorite/i }))

    expect(handleRemoveFavorite).toHaveBeenCalledTimes(1)
  })
})
