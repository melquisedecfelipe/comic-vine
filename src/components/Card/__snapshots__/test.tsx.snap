// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the card 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: Barlow,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
  font-size: 16px;
  padding: 0 16px;
  cursor: pointer;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  min-height: 40px;
  max-height: 40px;
  color: #FFFFFF;
  background: #000000;
}

.c2:focus {
  opacity: 0.75;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  height: 100%;
  background: #FFFFFF;
  border-bottom: 1px solid #F0ECFC;
  padding: 16px 0;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c0 img {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c0:hover img {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c1 {
  position: relative;
  height: 100%;
  width: 175px;
}

.c1 > button {
  position: absolute;
  top: 0;
  right: 0;
}

.c1 > button > svg {
  width: 20px;
  height: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c4 small {
  font-size: 12px;
  color: #707372;
  margin: 4px 0;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.c4 h3 {
  font-family: Barlow,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
  font-size: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.c4 p {
  font-size: 16px;
  color: #707372;
  margin: 12px 0;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
}

@media (min-width:768px) {
  .c0 {
    display: grid;
    grid-template-columns: 175px 1fr;
  }
}

@media (min-width:768px) {
  .c4 {
    margin-left: 16px;
  }
}

<div>
  <div
    class="c0"
  >
    <aside
      class="c1"
    >
      <img
        alt="Roy Mustang"
        height="250"
        layout="responsive"
        src="https://comicvine.gamespot.com/a/uploads/scale_small/11117/111178336/5581737-roy_mustang_by_norygunz-d4rd8xl.png"
        width="175"
      />
      <button
        aria-label="Add Favorite"
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.595a5.904 5.904 0 0 0-3.996-1.558 5.942 5.942 0 0 0-4.213 1.758c-2.353 2.363-2.352 6.059.002 8.412l7.332 7.332c.17.299.498.492.875.492a.99.99 0 0 0 .792-.409l7.415-7.415c2.354-2.354 2.354-6.049-.002-8.416a5.938 5.938 0 0 0-4.209-1.754A5.906 5.906 0 0 0 12 4.595zm6.791 1.61c1.563 1.571 1.564 4.025.002 5.588L12 18.586l-6.793-6.793c-1.562-1.563-1.561-4.017-.002-5.584.76-.756 1.754-1.172 2.799-1.172s2.035.416 2.789 1.17l.5.5a.999.999 0 0 0 1.414 0l.5-.5c1.512-1.509 4.074-1.505 5.584-.002z"
          />
        </svg>
      </button>
    </aside>
    <section
      aria-label="Character"
      class="c4"
      role="link"
    >
      <span>
        <small>
          Sep 29, 1886
        </small>
        <h3>
          Roy Mustang
        </h3>
      </span>
      <p>
        Roy Mustang, the Flame Alchemist, is the Colonel in Central and has jurisdiction over most alchemist there. Known for his heroism in Ishval and his amazing skill with his Ignition Cloth Gloves, Roy has great ambitions to become the next Fuhrer of Amestris.
      </p>
      <small>
        Aliases: 
        The Flame Alchemist Colonel Mustang General Mustang
      </small>
    </section>
  </div>
</div>
`;
